open-api/rest-catalog-open-api.yaml

```diff
@@ -3260,6 +3260,71 @@ components:
           additionalProperties:
             type: string
 
+    ReadRestrictions:
+      type: object
+      description: >
+          Read restrictions for a table, including projection and row filter expressions, according to the current schema.
+
+          A client MUST enforce the restrictions defined in this object when reading data
+          from the table.
+
+          These restrictions apply only to the authenticated principal, user, or account
+          associated with the client. They MUST NOT be interpreted as global policy and
+          MUST NOT be applied beyond the entity identified by the Authentication header
+          (or other applicable authentication mechanism).
+      properties:
+        required-projection:
+          description: >
+            A list of projections that MUST be applied prior to any query-specified
+            projections.
+            If the required-projection property is absent or empty, no mandatory projection applies,
+            and a reader MAY project any subset of columns of the table, including all columns.
+
+            1. A reader MUST project only columns listed in the required-projection.
+              - If a listed column has a transform, the reader MUST apply it and replace
+                all references to the underlying column with the transformed value
+                (for example, truncate(4, cc) MUST be projected as truncate(4, cc) AS cc,
+                and all references to cc during query evaluation MUST resolve to this alias).
+              - If a listed column has no transform, the reader MUST read it as-is.
+              - Columns not listed in the required-projection MUST NOT be read.
+
+            2. A column MUST appear at most once in the required-projection.
+      
+            3. A projection entry MUST reference either the column itself or exactly one
```

Comment:
I'm trying to understand how this is different from 2. and 4.? Do you have an example in mind?